### PR TITLE
change AbsoluteEntityCell usage to BatchId

### PR DESCRIPTION
**Pull request by killzoms**
TODO:
- [ ] Figure out why entities partially spawn on server reload
- [ ] Fix entities partially spawning on server reload

### CODE REVIEW

File: NitroxPatcher/Patches/Dynamic/LargeWorldStreamer_LoadBatchTask_Execute_Patch.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+using NitroxClient.GameLogic;
+using NitroxModel.Helper;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+public class LargeWorldStreamer_LoadBatchTask_Execute_Patch : NitroxPatch, IDynamicPatch
+{
+    public static readonly MethodInfo TARGET_METHOD = Reflect.Method((LargeWorldStreamer.LoadBatchTask t) => t.Execute());
+
+    public static bool Prefix(BatchCells ___batchCells)
```
Comment (Jannify) [inline NitroxPatcher/Patches/Dynamic/LargeWorldStreamer_LoadBatchTask_Execute_Patch.cs:12]:
```suggestion
    public static void Prefix(BatchCells ___batchCells)
```

File: NitroxPatcher/Patches/Dynamic/LargeWorldStreamer_UnloadBatch_Patch.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+using NitroxClient.GameLogic;
+using NitroxModel.Helper;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+public class LargeWorldStreamer_UnloadBatch_Patch : NitroxPatch, IDynamicPatch
+{
+    public static readonly MethodInfo TARGET_METHOD = Reflect.Method((LargeWorldStreamer t) => t.UnloadBatch(default(Int3)));
+
+    public static bool Prefix(Int3 index)
```
Comment (Jannify) [inline NitroxPatcher/Patches/Dynamic/LargeWorldStreamer_UnloadBatch_Patch.cs:12]:
```suggestion
    public static void Prefix(Int3 index)
```

File: NitroxServer/Communication/Packets/Processors/BatchVisibilityChangedProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+using System.Linq;
+using NitroxModel.DataStructures;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.DataStructures.GameLogic.Entities;
+using NitroxModel.Packets;
+using NitroxServer.Communication.Packets.Processors.Abstract;
+using NitroxServer.GameLogic.Entities;
+
+namespace NitroxServer.Communication.Packets.Processors;
+public class BatchVisibilityChangedProcessor : AuthenticatedPacketProcessor<BatchVisibilityChanged>
```
Comment (Jannify) [inline NitroxServer/Communication/Packets/Processors/BatchVisibilityChangedProcessor.cs:11]:
```suggestion
namespace NitroxServer.Communication.Packets.Processors;

public class BatchVisibilityChangedProcessor : AuthenticatedPacketProcessor<BatchVisibilityChanged>
```

File: NitroxServer/Communication/Packets/Processors/BatchVisibilityChangedProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    public BatchVisibilityChangedProcessor(EntitySimulation entitySimulation, WorldEntityManager worldEntityManager)
+    {
+        this.entitySimulation = entitySimulation;
+        this.worldEntityManager = worldEntityManager;
+    }
+
+    public override void Process(BatchVisibilityChanged packet, Player player)
+    {
+        SendBatchEntities(player, packet.Added);
```
Comment (Jannify) [inline NitroxServer/Communication/Packets/Processors/BatchVisibilityChangedProcessor.cs:24]: Why is this private method needed and not inline?

File: NitroxServer/GameLogic/Entities/Spawning/BatchEntitySpawner.cs [hunk 5]
```diff
@@ -69,11 +69,19 @@ public List<NitroxInt3> SerializableParsedBatches
...
+            return parsedBatches.Contains(batchId);
+        }
+    }
+
     public List<Entity> LoadUnspawnedEntities(NitroxInt3 batchId, bool fullCacheCreation = false)
     {
         lock (parsedBatches)
         {
-            if (parsedBatches.Contains(batchId))
+            if (IsBatchSpawned(batchId))
```
Comment (Jannify) [inline NitroxServer/GameLogic/Entities/Spawning/BatchEntitySpawner.cs:84]: would leave it with `parsedBatches.Contains` to not double lock

**Comment by Sunrunner37** [comment]
One thing we also want to add to the PR: we need to figure out why entities partially spawn upon server reload.

**Comment by killzoms** [comment]
yes, indeed

**Review by Sunrunner37 — APPROVED** [review]
We discussed in discord that the outstanding items are not actually a regression.  Now that we moved to batches, the spawn area just has a lot more items.  Long-term we should look to holding the wait screen longer while the initial batches finish entity spawning.  However, that is outside the scope of this specific PR and I don't want to block an improved experience on master.

**Pull request merged**